WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java

```diff
@@ -183,9 +183,14 @@ protected void onPostExecute(Boolean postUploadedSuccessfully) {
         @Override
         protected void onCancelled(Boolean aBoolean) {
             super.onCancelled(aBoolean);
-            mPostUploadNotifier.updateNotificationWithError(mErrorMessage, mIsMediaError, mPost.isPage(),
-                    mErrorUnavailableVideoPress);
-            WordPress.postUploadFailed(mPost.getLocalTableBlogId());
+            // mPostUploadNotifier and mPost can be null if onCancelled is called before doInBackground
+            if (mPostUploadNotifier != null) {
+                mPostUploadNotifier.updateNotificationWithError(mErrorMessage, mIsMediaError, mPost.isPage(),
```

Comment:
Probably not: https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2479-npe-postuploadservice-oncancelled/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java#L199-L201

but I'll join both `if`s.